libbeat/docs/breaking.asciidoc

```diff
@@ -9,9 +9,152 @@ changes, but there are breaking changes between major versions (e.g. 5.x to
+searches in Elasticsearch still work as expected. 
+
+*What do you need to change?*
+
+If you searched for the `host` field previously, you now need to search for the
+`host` and `host.name` fields. Or you can adjust your queries to use
+`beat.hostname`, instead of `host`, to work across all indices. 
+
+// REVIEWERS: Would it better just to tell users to uses to change their
+// searches to use `beat.hostname` instead of `host`?
+
+If you have visualizations in Kibana that reference the `host` field, export the
+objects by using the Kibana API or UI, change any aggregation references to use
+`beat.hostname` instead of `host`, and then re-import the objects into Kibana.
+
+// REVIEWERS: Are the above details correct? Got them from a comment Pius made
+// on ruflin's doc.
+
+//TODO: Add links to the Kibana docs for exporting/importing objects.
+
+[[custom-template-non-versioned-indices]]
+===== Use case: You use a custom template and your indices are not versioned
+
+Mapping conflicts are likely in this use case because two different Beats
+versions (6.2 and 6.3) are sending data to the same index. For 6.2, Logstash
+adds the default `host` field, and for 6.3, Beats adds the `host.name` field,
+which results in a mapping conflict.
+
+*What do you need to change?*
+
+To resolve the mapping issue, do *one* of the following:
+
+* Use versioned indices to prevent the mapping conflict. In the Logstash
+pipeline configuration, set `manage_template => false` and use an index naming
+pattern that includes `[version]` metadata:
++
+[source,yaml]
+----
+manage_template => false
+index => "%{[@metadata][beat]}-%{[@metadata][version]}-%{+YYYY.MM.dd}"
+----
++
+For more information, see the documentation for the
+{logstash-ref}/plugins-inputs-beats.html[Beats input plugin].
+
+* Or, in the Beats config file, configure Beats to drop all `host.*` fields: 
++
+[source,yaml]
+----
+  processors:
+   - drop_fields:
+      fields: ["host"]
+----
++
+--
+IMPORTANT: If you drop the `host` fields, you cannot use the `add_host_metadata`
+processor.
+
+--
++
+With this configuration, Beats drops the `host` fields before sending the
+event to Logstash, and Logstash adds a default `host` field, as it did with
+previous Beats versions.
+
+[[beats-template-non-versioned-indices]]
+===== Use case: You use the Beats index template and your indices are not versioned
+
+In this use case, you load the Beats index template manually into Elasticsearch,
+and send your data through Logstash, but you don’t use the versioned index
+pattern to create versioned indices.
+
+You cannot resolve the problem by dropping the `host.*` fields, as described
+earlier, because Logstash will add a default `host` field, resulting in a
+mapping conflict because the index template defines the `host` field as an
+object. 
+
+*What do you need to change?*
+
+To resolve the mapping issue, do *one* of the following:
+
+// TOOD: Should add an example of the LS filter here, don't you think?
```

Comment:
++

@ph Could you provide one? 

There is a bit a difference if `host.*` is dropped by Beats or LS. If it is dropped by Beats, LS will add `host` field. If it's dropped by LS, there will be no host field at all in the final event. I wonder if we should mention that difference.

Reply:
@ruflin @dedemorton  Yes, the following example will bring to pre 6.3.0. 

```
mutate {
  remove_field => [ "[host]" ]
}
mutate {
  add_field => {
    "host" => "%{[beat][hostname]}"
   }
}
```

Reply:
@ruflin but taking the content of `[beat][hostname]` for host will result in the same behavior as the input had. WDYT?

Reply:
I will do a check, but the _add_field_ should replace the original value.

Reply:
@ph See my comment ^^

Reply:
@dedemorton this is perfectly valid, thanks for  that :)
```
mutate {
  add_field => {
    "host" => "%{[host][name]}"
   }
}```